pilco/controllers.py

```diff
@@ -20,33 +24,36 @@ def squash_sin(m, s, max_action=None):
     else:
         max_action = max_action * tf.ones((1,k), dtype=float_type)
 
-    M = max_action * tf.exp(-tf.diag_part(s) / 2) * tf.sin(m)
+    M = max_action * tf.exp(-tf.linalg.diag_part(s) / 2) * tf.sin(m)
 
-    lq = -( tf.diag_part(s)[:, None] + tf.diag_part(s)[None, :]) / 2
+    lq = -( tf.linalg.diag_part(s)[:, None] + tf.linalg.diag_part(s)[None, :]) / 2
     q = tf.exp(lq)
     S = (tf.exp(lq + s) - q) * tf.cos(tf.transpose(m) - m) \
         - (tf.exp(lq - s) - q) * tf.cos(tf.transpose(m) + m)
     S = max_action * tf.transpose(max_action) * S / 2
 
-    C = max_action * tf.diag( tf.exp(-tf.diag_part(s)/2) * tf.cos(m))
+    C = max_action * tf.linalg.diag( tf.exp(-tf.linalg.diag_part(s)/2) * tf.cos(m))
     return M, S, tf.reshape(C,shape=[k,k])
 
 
-class LinearController(gpflow.Parameterized):
-    def __init__(self, state_dim, control_dim, max_action=None):
-        gpflow.Parameterized.__init__(self)
-        self.W = gpflow.Param(np.random.rand(control_dim, state_dim))
-        self.b = gpflow.Param(np.random.rand(1, control_dim))
+class LinearController(gpflow.Module):
+    def __init__(self, state_dim, control_dim, max_action=1.0, t=None):
+        gpflow.Module.__init__(self)
+        self.W = Parameter(np.random.rand(control_dim, state_dim))
+        self.b = Parameter(np.random.rand(1, control_dim))
+        if t is None:
+            self.t = 0.0
+        else:
+            self.t = tf.constant(t, shape=[1, state_dim])
         self.max_action = max_action
 
-    @gpflow.params_as_tensors
     def compute_action(self, m, s, squash=True):
         '''
         Simple affine action:  M <- W(m-t) - b
         IN: mean (m) and variance (s) of the state
         OUT: mean (M) and variance (S) of the action
         '''
-        M = m @ tf.transpose(self.W) + self.b # mean output
+        M = (m - self.t) @ tf.transpose(self.W) + self.b # mean output
```

Comment:
How does `self.t` help, since an offset is already present (`self.b`)?

Reply:
Yeah looks superfluous to me too.